store/sqlstore/channel_store.go

```diff
@@ -2350,6 +2346,7 @@ func (s SqlChannelStore) ClearAllCustomRoleAssignments() store.StoreChannel {
 				result.Err = model.NewAppError("SqlChannelStore.ClearAllCustomRoleAssignments", "store.sql_channel.clear_all_custom_role_assignments.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
 				return
 			}
+			defer transaction.Rollback()
```

Comment:
This one isn't quite correct, since the rollbacks won't occur until the method returns, but the semantics are probably meant to commit or rollback within each pass of the `for` loop. Perhaps factor out the body of the for loop as a method to adopt this paradigm instead?

Reply:
Thanks again for the rapid and thorough review, @lieut-data!

I agree. I can indeed factor the `for` loop body in a separate method to have the rollback executed after each iteration. I'll just wait for @jespino's opinion before committing these changes.

Reply:
Agree. Either factor it out into a func, or make/keep rollbacks explicit, perhaps with a comment indicating that `defer ...Rollback()` is purposely missing?  I do not have a preference.